gammapy/spectrum/fit.py

```diff
@@ -153,90 +88,91 @@ def fit_range(self):
     def fit_range(self, fit_range):
         self._fit_range = fit_range
 
-    @property
-    def pha_list(self):
-        """Comma-separate list of PHA files"""
-        file_list = [str(o.phafile) for o in self.obs_list]
-        ret = ','.join(file_list)
-        return ret
-
     def __str__(self):
         """String repr"""
         ss = 'Model\n'
         ss += str(self.model)
         return ss
 
-    def run(self, method='sherpa', outdir=None):
+    def fit(self):
+        """Fit spectrum"""
+        from sherpa.fit import Fit
+        from sherpa.models import ArithmeticModel, SimulFitModel
+        from sherpa.astro.instrument import Response1D
+        from sherpa.data import DataSimulFit
+
+        # Translate model to sherpa model if necessary
+        if isinstance(self.model, models.SpectralModel):
+            model = self.model.to_sherpa()
+        else:
+            model = self.model
+
+        if not isinstance(model, ArithmeticModel):
+            raise ValueError('Model not understood: {}'.format(model))
+
+        # Make model amplitude O(1e0)
+        val = model.ampl.val * self.FLUX_FACTOR ** (-1)
+        model.ampl = val
+
+        if self.fit_range is not None:
+            log.info('Restricting fit range to {}'.format(self.fit_range))
+            fitmin = self.fit_range[0].to('keV').value
+            fitmax = self.fit_range[1].to('keV').value
+
+        # Loop over observations
+        pha = list()
+        folded_model = list()
+        nobs = len(self.obs)
+        for ii in range(nobs):
+            temp = self.obs[ii].to_sherpa()
+            if self.fit_range is not None:
+                temp.notice(fitmin, fitmax)
+                temp.get_background().notice(fitmin, fitmax)
+            temp.ignore_bad()
+            temp.get_background().ignore_bad()
+            pha.append(temp)
+            # Forward folding
+            resp = Response1D(pha[ii])
+            folded_model.append(resp(model) * self.FLUX_FACTOR)
+
+        data = DataSimulFit('simul fit data', pha)
+        fitmodel = SimulFitModel('simul fit model', folded_model)
+
+        fit = Fit(data, fitmodel, self.statistic)
+        fitresult = fit.fit()
+        log.info(fitresult)
```

Comment:
Maybe remove those `log.info` calls or change to `log.debug`?
Maybe instead give the caller access to the Sherpa `fit` or `fitresult` info by returning it from this method or storing it on the class?